bindings/rust/s2n-tls/src/session_ticket.rs

```diff
@@ -0,0 +1,57 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+use core::ptr::NonNull;
+use std::time::Duration;
+
+use s2n_tls_sys::*;
+
+use crate::{
+    connection::Connection,
+    error::{Error, Fallible},
+};
+
+/// A trait to retrieve session tickets from the connection
+pub trait SessionTicketCallback {
```

Comment:
This needs to be `Send` + `Sync` since `Config` is marked as `Send` + `Sync` and can be referenced across multiple threads. If the other callbacks aren't marked as such, that's a bug.

Reply:
😬 Yeah looks like none of them are. Should I add that to them?

Reply:
https://github.com/aws/s2n-tls/issues/4131